keps/sig-api-machinery/1872-manifest-based-admission-webhooks/README.md

```diff
@@ -155,22 +138,66 @@ plugins:
 ### Reconfiguring manifest file
 
 The manifest file is watched so that the webhook configuration can be
-dynamically changed by editing the contents of the file.
+dynamically changed by editing the contents of the file. In addition to watching
+the file for changes, we would also read the file periodically checking for
+changes.
+
+#### Behavioral details of reconfiguration
+This section details the behavior of the mechanism when encountering non-optimal
+situations.
+
+In cases where:
+* the given manifest file can no longer be found or
+* attempts at reading the manifest file result in an error or
+* the contents of the manifest file can no longer be parsed or
+* validation errors are encountered when parsing the webhook configuration of
+the manifest file,
+
+the previously successfully loaded set of webhooks will continue to be invoked
+when necessary. An error would be surfaced through a metric (See the metrics
+section). If the file or a valid version of it re-appears, the new file will to
+loaded and if valid, the webhooks in this file would be used for future
+invocations. Also, the error metric would be unset.
 
-```
-<<[UNRESOLVED]>>
-need to define behavior in the following cases:
-1. file goes missing
-2. file cannot be read
-3. parse error reading file contents
-4. some webhook configurations listed in the file do not pass validation
-<<[/UNRESOLVED]>>
-```
+### Metrics and audit annotations
+For the administrator to uniquely monitor manifest based webhooks, two
+additional metrics would be added at [stability level `ALPHA`](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md#stability-classes)
+to the existing set of metrics.
+
+a. Webhook metadata: This metric would contain metadata of each webhook invoked,
+both API based and manifest based. The metadata includes the webhook type
+(mutating/validating) and if it is manifest based.
```

Comment:
detail the name and labels of the metric?

Reply:
also, add the metric to the `metrics` section of the kep.yaml file (same for all the metrics)